libbpfgo.go

```diff
@@ -1331,6 +1384,61 @@ func (p *BPFProg) AttachCgroup(cgroupV2DirPath string) (*BPFLink, error) {
 	return bpfLink, nil
 }
 
+// AttachCgroupLegacy attaches the BPFProg to a cgroup described by given fd.
+// It first tries to use the most recent attachment method and, if that does
+// not work, instead of failing it tries the legacy way: to attach the cgroup
+// eBPF program without previously creating a link. Related to upstream kernel
+// commit af6eea57437a ("bpf: Implement bpf_link-based cgroup BPF program
+// attachment"). This allows attaching cgroup eBPF ingress/egress in older
+// kernels.
+func (p *BPFProg) AttachCgroupLegacy(cgroupV2DirPath string, attachType BPFAttachType) (*BPFLink, error) {
+	// bpfLink, err := p.AttachCgroup(cgroupV2DirPath)
```

Comment:
I'll un-comment this in the final merge (it is like that so anyone testing this code will correctly test it in newer kernels as well). I'll check if 5.4 kernels are good before merging as well.